brave/brave-core
# Switch vpn protocols during the runtime

fix https://github.com/brave/brave-browser/issues/35681

`BraveVPNOSConnectionAPI` becomes concrete class and it will have
proper `ConnectionAPIImpl` instance for selected vpn protocol.


Resolves 



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

Another test plan from issue.
1. Launch brave and set purchased state
2. Connect vpn and check it's connected with current settings option(wg or ikev2)
3. Disconnect
4. Toggle wireguard settings option
5. Connect vpn again and check it's connected with another option
6. Relaunch browser and connect with existing setting
7. Check connection is established with current setting
8. Disconnect and toggle wireguard settings and connect again
9. Check connection is made with toggled option
